bundle/terranova/tnresources/adapter.go

```diff
@@ -25,12 +25,37 @@ type IResource interface {
 	// Example: func (*ResourceJob) PrepareConfig(input *resources.Job) *jobs.JobSettings
 	PrepareConfig(input any) any
 
+	// DoRefresh reads and returns remote state from the backend. The return type defines schema for remote field resolution.
+	// Example: func (r *ResourceJob) DoRefresh(ctx context.Context, id string) (*jobs.Job, error) {
+	DoRefresh(ctx context.Context, id string) (remoteState any, e error)
+
 	// DoDelete deletes the resource.
 	// Example: func (r *ResourceJob) DoDelete(ctx context.Context, id string) error {
 	DoDelete(ctx context.Context, id string) error
 
+	// [Optional] RecreateFields returns a list of fields that will cause resource recreation if changed
+	// Example: func (r *ResourceJob) RecreateFields() []string { return []string{"name", "type"} }
+	RecreateFields() []string
+
+	// [Optional] ClassifyChanges provides non-default change classification.
+	// Default is to consider any change "an update" (RecreateFields handled separately).
+	// Example: func (r *ResourceJob) ClassifyChanges(changes []structdiff.Change) deployplan.ActionType { return deployplan.ActionUpdate }
+	ClassifyChanges(changes []structdiff.Change) deployplan.ActionType
+}
+
+// This interface represents adapted interface to call a resource.
+// The resources don't actually implement this interface, but rather a subset of functions
+// with different signatures: any is replaced with concrete types.
+type IResourceAdapterNoRefresh interface {
+	// Any field named config below have the same type as return value of PrepareConfig.
+	// Any field named remoteState below has the same type as return value of DoRefresh.
```

Comment:
Thought: since the `any`s below use the same type, can't we use a generic interface to capture the type and make type assertions? It doesn't remove the need for call adapters because the generic interface needs to be made concrete, but it can enforce that resources implement one of these at compile time.